pkg/services/gitopswriter/gitopswriter.go

```diff
@@ -144,11 +149,77 @@ func (dw *gitOpsDirectoryWriterSvc) RemoveApplication(ctx context.Context, app m
 	return dw.RepoWriter.CommitAndPush(ctx, RemoveCommitMessage)
 }
 
+func (dw *gitOpsDirectoryWriterSvc) AssociateCluster(
+	ctx context.Context,
+	cluster models.Cluster,
+	configURL gitproviders.RepoURL,
+	namespace string,
+	autoMerge bool) error {
+	auto, err := dw.Automation.GenerateClusterAutomation(ctx, cluster, configURL, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to generate cluster automation: %w", err)
+	}
+
+	manifests := auto.Manifests()
+
+	defaultBranch, err := dw.RepoWriter.GetDefaultBranch(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve default branch for repository: %w", err)
+	}
+
+	// store a .keep file in the user dir
+	userKeep := automation.AutomationManifest{
+		Path:    filepath.Join(getUserDirRepoPath(cluster.Name), ".keep"),
+		Content: []byte(strconv.AppendQuote(nil, "# keep")),
+	}
+
+	manifests = append(manifests, userKeep)
+
+	dw.Logger.Actionf("Associating cluster %q", cluster.Name)
+
+	if autoMerge {
+		remover, repoDir, err := dw.RepoWriter.CloneRepo(ctx, defaultBranch)
+		if err != nil {
+			return fmt.Errorf("failed to clone repo: %w", err)
+		}
+
+		defer remover()
+
+		if err := dw.RepoWriter.WriteAndMerge(ctx, repoDir, ClusterCommitMessage, manifests); err != nil {
+			return fmt.Errorf("failed writing automation to disk: %w", err)
+		}
+	} else {
+		files := []gitprovider.CommitFile{}
+
+		for _, manifest := range manifests {
+			manifestPath := manifest.Path
+			content := string(manifest.Content)
+
+			files = append(files, gitprovider.CommitFile{Path: &manifestPath, Content: &content})
+		}
+
+		prInfo := gitproviders.PullRequestInfo{
+			Title:         fmt.Sprintf("GitOps associate %s", cluster.Name),
+			Description:   fmt.Sprintf("Added yamls for %s", cluster.Name),
```

Comment:
```suggestion
			Description:   fmt.Sprintf("Add gitops automation manifests for cluster %s", cluster.Name),
```

This message is a bit clearer when using the "If applied, this commit will .."  from this post - https://cbea.ms/git-commit/